File: legacy/node/src/animations/spinner.js
const chalk = require('chalk');
const gradient = require('gradient-string');

// Animation de la roue de la fortune
async function spinWheel(users, selectedUser, chore) {
  const frames = ['🎰', '🎲', '🎯', '🎪', '🎨', '🎭', '🎬', '🎪'];
  const spinDuration = 2000; // 2 secondes
  const frameDelay = 80;
  const iterations = Math.floor(spinDuration / frameDelay);
  
  // Effacer la ligne pour l'animation
  process.stdout.write('\n');
  
  for (let i = 0; i < iterations; i++) {
    const randomUser = users[Math.floor(Math.random() * users.length)];
    const frame = frames[i % frames.length];
    const progress = i / iterations;
    
    // Créer l'effet de ralentissement
    const isSlowingDown = progress > 0.7;
    
    // Texte avec gradient
    let displayText;
    if (progress < 0.3) {
      displayText = gradient.rainbow(`${frame} ${chore.padEnd(15)} → ${randomUser}...`);
    } else if (progress < 0.7) {
      displayText = gradient.cristal(`${frame} ${chore.padEnd(15)} → ${randomUser}...`);
    } else {
      displayText = gradient.pastel(`${frame} ${chore.padEnd(15)} → ${randomUser}...`);
    }
    
    // Afficher
    process.stdout.write('\r' + displayText);
    
    // Attendre
    const delay = isSlowingDown ? frameDelay * (1 + progress * 2) : frameDelay;
    await new Promise(resolve => setTimeout(resolve, delay));
  }
  
  // Animation finale - révélation
  process.stdout.write('\r' + ' '.repeat(80) + '\r');
  
  // Effet de flash
  for (let i = 0; i < 3; i++) {
    if (i % 2 === 0) {
      process.stdout.write('\r' + gradient.rainbow(`✨ ${chore.padEnd(15)} → ${selectedUser} ✨`));
    } else {
      process.stdout.write('\r' + chalk.bold.yellow(`✨ ${chore.padEnd(15)} → ${selectedUser} ✨`));
    }
    await new Promise(resolve => setTimeout(resolve, 150));
  }
  
  // Résultat final avec effet
  process.stdout.write('\r' + ' '.repeat(80) + '\r');
  console.log(chalk.green('✓'), chalk.bold(chore.padEnd(15)), '→', gradient.rainbow.multiline(selectedUser));
  
  // Petite pause pour l'effet
  await new Promise(resolve => setTimeout(resolve, 300));
}

// Animation de titre
function showTitle() {
  console.log('');
  console.log(gradient.rainbow('╔═══════════════════════════════════════════════╗'));
  console.log(gradient.rainbow('║                                               ║'));
  console.log(gradient.rainbow('║        🎰  ROUE DE LA FORTUNE  🎰           ║'));
  console.log(gradient.rainbow('║          Répartition des corvées              ║'));
  console.log(gradient.rainbow('║                                               ║'));
  console.log(gradient.rainbow('╚═══════════════════════════════════════════════╝'));
  console.log('');
}

module.exports = {
  spinWheel,
  showTitle
};

File: chore_manager/animations/spinner.py
import random
import time
from typing import List

from rich.text import Text

from ..utils.console import get_console
from ..utils.gradient import rainbow_text, pastel_text, cristal_text


FRAMES = ["🎰", "🎲", "🎯", "🎪", "🎨", "🎭", "🎬", "🎪"]
SPIN_DURATION = 2.0
FRAME_DELAY = 0.08


def show_title() -> None:
    console = get_console()
    lines = [
        "╔═══════════════════════════════════════════════╗",
        "║                                               ║",
        "║        🎰  ROUE DE LA FORTUNE  🎰           ║",
        "║          Répartition des corvées              ║",
        "║                                               ║",
        "╚═══════════════════════════════════════════════╝",
    ]
    console.print("")
    for line in lines:
        console.print(rainbow_text(line))
    console.print("")


def spin_wheel(
    users: List[str],
    selected_user: str,
    chore: str,
    animate: bool = True,
) -> None:
    console = get_console()

    if not animate:
        console.print(
            Text("✓ ", style="green")
            + Text(chore.ljust(15), style="bold")
            + Text(" → ")
            + rainbow_text(selected_user)
        )
        return

    iterations = int(SPIN_DURATION / FRAME_DELAY)
    console.print("")

    for i in range(iterations):
        random_user = random.choice(users)
        frame = FRAMES[i % len(FRAMES)]
        progress = i / iterations
        is_slowing = progress > 0.7

        line = f"{frame} {chore.ljust(15)} → {random_user}..."
        if progress < 0.3:
            text = rainbow_text(line)
        elif progress < 0.7:
            text = cristal_text(line)
        else:
            text = pastel_text(line)

        console.print(text, end="\r", soft_wrap=True)

        delay = FRAME_DELAY * (1 + progress * 2) if is_slowing else FRAME_DELAY
        time.sleep(delay)

    console.print(" " * 80, end="\r")

    for i in range(3):
        if i % 2 == 0:
            text = rainbow_text(f"✨ {chore.ljust(15)} → {selected_user} ✨")
        else:
            text = Text(f"✨ {chore.ljust(15)} → {selected_user} ✨", style="bold yellow")
        console.print(text, end="\r", soft_wrap=True)
        time.sleep(0.15)

    console.print(" " * 80, end="\r")
    console.print(
        Text("✓ ", style="green")
        + Text(chore.ljust(15), style="bold")
        + Text(" → ")
        + rainbow_text(selected_user)
    )
    time.sleep(0.3)
